libsupport/include/katana/ArrowVisitor.h

```diff
@@ -123,56 +105,122 @@ struct BuilderVisitorBaseType {
 /// };
 /// Arrow's type_traits.h offers tools to use SFINAE to differentiate
 /// types and write appropriate Call functions for each type
-template <class VisitorBaseType, class VisitorType>
-katana::Result<typename std::decay_t<VisitorType>::ReturnType>
-VisitArrowInternal(
-    typename VisitorBaseType::ParamBase param, VisitorType&& visitor) {
-  switch (VisitorBaseType::Type(param)->id()) {
+
+// The Wrapper struct is used to keep track of the ArrowType for each argument
+// as we recursively cast each argument. The ArrowTypes are then used as
+// template arguments to the Call method of the Visitor class
+template <class... ArrowTypes>
+struct Wrapper {
```

Comment:
Can you please give this type a better name?

"Wrapper" is pretty common idea and although this is in the internal namespace, it is still widely visible and can collide with other uses of "Wrapper".

How about "ArrowWrapper" at least?